src/storage/mod.rs

```diff
@@ -117,10 +166,73 @@ impl Storage {
         }
     }
 
+    fn max_file_size_for(&self, path: &str) -> usize {
+        if path.ends_with(".html") {
+            self.max_file_size_html
+        } else {
+            self.max_file_size
+        }
+    }
+
+    pub(crate) fn fetch_rustdoc_file(
+        &self,
+        name: &str,
+        version: &str,
+        path: &str,
+        archive_storage: bool,
+    ) -> Result<Blob, Error> {
+        Ok(if archive_storage {
+            let archive = format!("rustdoc/{0}/rustdoc-{0}-{1}.zip", name, version);
```

Comment:
Remember to update this if you change the storage path above.
```suggestion
            let archive = format!("rustdoc/{0}/{1}.zip", name, version);
```